trino/trino.go

```diff
@@ -976,11 +1022,34 @@ func (qr *driverRows) fetch(allowEOF bool) error {
 	qr.rowindex = 0
 	qr.data = qresp.Data
 	qr.nextURI = qresp.NextURI
+
+	qrStats := QueryProgressCallbackResponse{
+		ID:         qresp.ID,
+		QueryStats: qresp.Stats,
+		Error:      qresp.Error,
+	}
+
 	if len(qr.data) == 0 {
 		if qr.nextURI != "" {
+			if qr.stmt.conn.callback != nil {
+				if qrStats.QueryStats.State == "RUNNING" {
+					currentTime := time.Now()
+					diff := currentTime.Sub(qr.stmt.conn.callbackPeriod.LastCallbackTime).Milliseconds()
+					period := qr.stmt.conn.callbackPeriod.Period.Milliseconds()
+					// Check if period has passed
+					if diff >= period {
```

Comment:
Why you cast it to milliseconds? Isn't `time.Time` comparable?

Reply:
I i don't quite understand how to with simply using `time.Time` comparable since time.Sub returns time.Duration.
1. need to find time elapsed since last call back time
2. compare with period
3. if elapsed time exceeds period, update progress

Reply:
Oh sorry, I meant comparing `time.Duration`. You don't need to extract milliseconds. This program:
```go
package main

import (
	"fmt"
	"time"
)

func main() {
	now := time.Now()
	before := time.Date(2009, 11, 10, 21, 0, 0, 0, time.UTC)
	fmt.Printf("now: %s,\nbefore: %s\n", now, before)
	diff := now.Sub(before)
	period := 1 * time.Hour
	if diff > period {
		fmt.Printf("%s > %s\n", diff, period)
	} else {
		fmt.Printf("%s <= %s\n", diff, period)
	}
}
```
prints:
```
now: 2009-11-10 23:00:00 +0000 UTC m=+0.000000001,
before: 2009-11-10 21:00:00 +0000 UTC
2h0m0s > 1h0m0s
```
https://go.dev/play/p/Fsi0p4YGKNx